src/basic/stat-util.c

```diff
@@ -408,7 +408,7 @@ bool stat_inode_unmodified(const struct stat *a, const struct stat *b) {
         return a && b &&
                 (a->st_mode & S_IFMT) != 0 && /* We use the check for .st_mode if the structure was ever initialized */
                 ((a->st_mode ^ b->st_mode) & S_IFMT) == 0 &&  /* same inode type */
-                a->st_mtime == b->st_mtime &&
+                (memcmp(&a->st_mtim, &b->st_mtim, sizeof(a->st_mtim)) == 0) &&
```

Comment:
hmm, we can't use memcmp() for this, due to potential padding issues. i.e. there might be padding between the two fields of timespec in some arch, and we shouldn#t compare that. Hence, please change to:
```c
…
a->st_mtim.tv_sec == b->st_mtim.tv_sec &&
a->st_mtim.tv_nsec == b->st_mtim.tv_nsec &&
…
```

(and that's not just theory, there are common archs where the tv_sec fields is 32bit wide, but tv_nsec 64bit, which means 32bit padding between)

Reply:
Good point